### PR TITLE
Batch base and pr runs

### PR DESCRIPTION
**Pull request by sebastienros**
A single build is performed for base or head when multiple benchmarks or profiles are provided.

### CODE REVIEW

File: src/Microsoft.Crank.PullRequestBot/Program.cs [hunk 1]
```diff
@@ -563,79 +563,101 @@ private static async Task<IEnumerable<Result>> RunBenchmark(Command command)
...
-                    RunCrank(_configuration.Defaults, benchmark.Arguments, profile.Arguments, buildArguments, $@"--json ""{workspace}{BaseFilename}""");
+                    RunCrank(_configuration.Defaults, benchmark.Arguments, profile.Arguments, buildArguments, $@"--json ""{baseResultsFilename}""");
+                }
 
-                    await ProcessUtil.RunAsync("git", $@"fetch origin pull/{prNumber}/head", workingDirectory: cloneFolder, log: true);
-                    await ProcessUtil.RunAsync("git", $@"config user.name ""user""", workingDirectory: cloneFolder, log: true);
-                    await ProcessUtil.RunAsync("git", $@"config user.email ""user@company.com""", workingDirectory: cloneFolder, log: true);
-                    await ProcessUtil.RunAsync("git", $@"merge FETCH_HEAD", workingDirectory: cloneFolder, log: true);
+                await ProcessUtil.RunAsync("git", $@"fetch origin pull/{prNumber}/head", workingDirectory: cloneFolder, log: true);
+                //await ProcessUtil.RunAsync("git", $@"config user.name ""user""", workingDirectory: cloneFolder, log: true);
```
Comment (adityamandaleeka) [inline src/Microsoft.Crank.PullRequestBot/Program.cs:610]: Should these be removed?

File: src/Microsoft.Crank.PullRequestBot/Program.cs [hunk 2]
```diff
@@ -563,79 +563,101 @@ private static async Task<IEnumerable<Result>> RunBenchmark(Command command)
...
-                while (Directory.Exists(Path.Combine(workspace, folder))) { folder = command.PullRequest.Base.Repository.Name + counter++; }
+            var cloneFolder = Path.Combine(workspace, folder);
 
-                var cloneFolder = Path.Combine(workspace, folder);
+            var buildCommands = command.Components.SelectMany(b => _configuration.Components[b].Script.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries));
 
-                var buildCommands = command.Components.SelectMany(b => _configuration.Components[b].Script.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries));
+            // 1- Build the base branch once
+            // 2- Run all benchmarks on all profiles
+            // 3- Build the PR branch once
```
Comment (adityamandaleeka) [inline src/Microsoft.Crank.PullRequestBot/Program.cs:581]: Maybe add a step here about how we merge the PR branch onto main.

**Review by adityamandaleeka — APPROVED** [review]

File: src/Microsoft.Crank.PullRequestBot/Program.cs (hunk 1 above)
Comment (sebastienros) [reply src/Microsoft.Crank.PullRequestBot/Program.cs]: I am hesitant so I am putting them back

File: src/Microsoft.Crank.PullRequestBot/Program.cs (hunk 2 above)
Comment (sebastienros) [reply src/Microsoft.Crank.PullRequestBot/Program.cs]: Added

**Pull request merged**